scripts/zones/Mhaura/npcs/Celestina.lua

```diff
@@ -6,57 +6,55 @@
-            player:startEvent(127); -- Finish quest "The Sand Charm"
+    if player:getQuestStatus(OTHER_AREAS_LOG,tpz.quest.id.otherAreas.THE_SAND_CHARM) == QUEST_ACCEPTED then
+        if npcUtil.tradeHasExactly(trade, 13095) then
+            player:startEvent(127) -- Finish quest "The Sand Charm"
         end
     end
 
-    if (player:getQuestStatus(JEUNO,tpz.quest.id.jeuno.RIDING_ON_THE_CLOUDS) == QUEST_ACCEPTED and player:getCharVar("ridingOnTheClouds_3") == 5) then
-        if (trade:hasItemQty(1127,1) and trade:getItemCount() == 1) then -- Trade Kindred seal
-            player:setCharVar("ridingOnTheClouds_3",0);
-            player:tradeComplete();
-            player:addKeyItem(tpz.ki.SOMBER_STONE);
-            player:messageSpecial(ID.text.KEYITEM_OBTAINED,tpz.ki.SOMBER_STONE);
+    if player:getQuestStatus(JEUNO, tpz.quest.id.jeuno.RIDING_ON_THE_CLOUDS) == QUEST_ACCEPTED and player:getCharVar("ridingOnTheClouds_3") == 5 then
+        if npcUtil.tradeHasExactly(trade, 1127) then -- Trade Kindred seal
+            player:setCharVar("ridingOnTheClouds_3",0)
+            player:confirmTrade()
+            npcUtil.giveKeyItem(player, tpz.ki.SOMBER_STONE)
         end
     end
-
-end;
-
-function onTrigger(player,npc)
-
-    if (player:getCharVar("theSandCharmVar") == 3) then
-        player:startEvent(126,13095); -- During quest "The Sand Charm" - 3rd dialog
-    elseif (player:sendGuild(528,8,23,4)) then
-        player:showText(npc,ID.text.GOLDSMITHING_GUILD);
+end
+
+function onTrigger(player, npc)
+    if player:getCharVar("theSandCharmVar") == 3 then
+        player:startEvent(126, 13095) -- During quest "The Sand Charm" - 3rd dialog
+    else
+        local guildSkillId = tpz.skill.GOLDSMITHING
+        local stock = tpz.shop.generalGuildStock[guildSkillId]
+        tpz.shop.generalGuild(player, stock, guildSkillId)
     end
+end
 
+function onEventUpdate(player, csid, option)
 end;
 
-function onEventUpdate(player,csid,option)
-end;
-
-function onEventFinish(player,csid,option)
-
-    if (csid == 126 and option == 70) then
-        player:setCharVar("theSandCharmVar",4);
+function onEventFinish(player, csid, option)
+    if csid == 126 and option == 70 then
+        player:setCharVar("theSandCharmVar", 4)
     elseif (csid == 127) then
-        player:tradeComplete();
-        player:setCharVar("theSandCharmVar",0);
-        player:setCharVar("SmallDialogByBlandine",1);
-        player:addKeyItem(tpz.ki.MAP_OF_BOSTAUNIEUX_OUBLIETTE);
-        player:messageSpecial(ID.text.KEYITEM_OBTAINED,tpz.ki.MAP_OF_BOSTAUNIEUX_OUBLIETTE);
-        player:addFame(MHAURA,30);
-        player:completeQuest(OTHER_AREAS_LOG,tpz.quest.id.otherAreas.THE_SAND_CHARM);
+        player:confirmTrade()
+        npcUtil.completeQuest(player, OTHER_AREAS_LOG, tpz.quest.id.otherAreas.THE_SAND_CHARM, {
+            keyItem = tpz.ki.MAP_OF_BOSTAUNIEUX_OUBLIETTE,
+            fame_area = MHAURA,
+            fame = 30,
```

Comment:
fame defaults to 30, so this parameter isn't needed

Reply:
fixed thank you @wrenffxi  :cat: